added error codes detection

AnastasiyaTarasova opened the pull request.

AnastasiyaTarasova committed:
Added _NeonEvmErrorLogDecoder

neonlabstech commented:
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getSlot                           |   59756 |      0     |      0.176 |    0.000986378 |         0.001 |
| getAccountInfo                    |   54711 |      0     |      0.12  |    0.0018296   |         0.001 |
| getBlocks                         |   24434 |      0     |      0.06  |    0.00071245  |         0.001 |
| getBlock                          |   16011 |      0     |      0.06  |    0.00178914  |         0.002 |
| getMultipleAccounts               |   13486 |      0     |      0.079 |    0.00233101  |         0.001 |
| getTransaction                    |   12393 |      0     |      0.241 |    0.011261    |         0.004 |
| getLatestBlockhash                |   10894 |      0     |      0.111 |    0.00165201  |         0.001 |
| getBlockTime                      |    5935 |      0     |      0.067 |    0.000653917 |         0     |
| sendTransaction                   |    5072 |      0     |      0.117 |    0.00880225  |         0.004 |
| getBalance                        |    1761 |      0     |      0.076 |    0.00158035  |         0.001 |
| getSignatureStatuses              |    1124 |      0     |      0.091 |    0.00192082  |         0.001 |
| getVersion                        |     601 |      0     |      0.048 |    0.00177704  |         0     |
| isBlockhashValid                  |     544 |      0     |      0.024 |    0.000516544 |         0     |
| simulateTransaction               |     491 |      0     |      0.006 |    0.00179633  |         0.001 |
| getSignaturesForAddress           |      61 |      0     |      0.016 |    0.00198361  |         0.002 |
| requestAirdrop                    |      32 |      0     |      0.001 |    0.00084375  |         0.001 |
| getTokenAccountBalance            |      11 |      0     |      0.047 |    0.0125455   |         0.001 |
| getHealth                         |       3 |      0     |      0.001 |    0.000333333 |         0     |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |         0.001 |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |         0     |
| getClusterNodes                   |       1 |      0     |      0     |    0           |         0     |

</details>

neonlabstech commented:
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getSlot                           |   57583 |      0     |      0.272 |    0.0010456   |         0.001 |
| getAccountInfo                    |   54716 |      0     |      0.249 |    0.00211244  |         0.001 |
| getBlocks                         |   23816 |      0     |      0.147 |    0.000750924 |         0.001 |
| getBlock                          |   15618 |      0     |      0.065 |    0.00191747  |         0.002 |
| getMultipleAccounts               |   13415 |      0     |      0.184 |    0.00252449  |         0.001 |
| getTransaction                    |   12498 |      0     |      0.381 |    0.0129432   |         0.004 |
| getLatestBlockhash                |   10707 |      0     |      0.131 |    0.00192416  |         0.001 |
| getBlockTime                      |    6010 |      0     |      0.044 |    0.000723461 |         0     |
| sendTransaction                   |    5112 |      0     |      0.147 |    0.00956025  |         0.005 |
| getBalance                        |    1763 |      0     |      0.06  |    0.00242995  |         0.001 |
| getSignatureStatuses              |    1124 |      0     |      0.25  |    0.00236388  |         0.001 |
| getVersion                        |     602 |      0     |      0.104 |    0.00194186  |         0     |
| isBlockhashValid                  |     544 |      0     |      0.271 |    0.00102941  |         0     |
| simulateTransaction               |     491 |      0     |      0.036 |    0.0036945   |         0.003 |
| getSignaturesForAddress           |      61 |      0     |      0.003 |    0.00139344  |         0.001 |
| requestAirdrop                    |      32 |      0     |      0.002 |    0.00090625  |         0.001 |
| getTokenAccountBalance            |      12 |      0     |      0.068 |    0.006       |         0     |
| getHealth                         |       3 |      0     |      0.001 |    0.000666667 |         0.001 |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |         0.001 |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |         0.001 |
| getClusterNodes                   |       1 |      0.001 |      0.001 |    0.001       |         0.001 |

</details>

neonlabstech commented:
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getSlot                           |   63247 |      0     |      0.193 |    0.000914652 |        0.001  |
| getAccountInfo                    |   54810 |      0     |      0.114 |    0.00178486  |        0.001  |
| getBlocks                         |   26018 |      0     |      0.052 |    0.000715582 |        0.001  |
| getBlock                          |   16982 |      0     |      0.033 |    0.00174738  |        0.002  |
| getMultipleAccounts               |   13573 |      0     |      0.193 |    0.00212849  |        0.001  |
| getTransaction                    |   12540 |      0     |      0.118 |    0.00904258  |        0.004  |
| getLatestBlockhash                |   11310 |      0     |      0.09  |    0.00142732  |        0.001  |
| getBlockTime                      |    5876 |      0     |      0.051 |    0.000649421 |        0      |
| sendTransaction                   |    5126 |      0     |      0.33  |    0.00914768  |        0.004  |
| getBalance                        |    1764 |      0     |      0.06  |    0.00177268  |        0.001  |
| getSignatureStatuses              |    1124 |      0     |      0.036 |    0.00138256  |        0.001  |
| getVersion                        |     588 |      0     |      0.035 |    0.00142177  |        0      |
| isBlockhashValid                  |     544 |      0     |      0.024 |    0.000612132 |        0      |
| simulateTransaction               |     491 |      0     |      0.006 |    0.00194908  |        0.002  |
| getSignaturesForAddress           |      67 |      0     |      0.005 |    0.0021194   |        0.002  |
| requestAirdrop                    |      32 |      0     |      0.001 |    0.000875    |        0.001  |
| getTokenAccountBalance            |       5 |      0     |      0.024 |    0.0078      |        0.001  |
| getHealth                         |       4 |      0     |      0.001 |    0.0005      |        0.0005 |
| getFirstAvailableBlock            |       2 |      0     |      0.001 |    0.0005      |        0.0005 |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |        0.001  |
| getClusterNodes                   |       1 |      0     |      0     |    0           |        0      |

</details>

AnastasiyaTarasova committed:
Added EvmLogDecoder decode

AnastasiyaTarasova committed:
wip

AnastasiyaTarasova committed:
bugfix

AnastasiyaTarasova committed:
test edited

AnastasiyaTarasova committed:
NeonSolTxListSender changes

AnastasiyaTarasova committed:
SolTxSig.default fix

AnastasiyaTarasova committed:
from_raw, decode, to_clean_copy fixes

AnastasiyaTarasova committed:
decode calling fix

AnastasiyaTarasova committed:
bugfix

AnastasiyaTarasova committed:
Added error to tx_error_list

AnastasiyaTarasova committed:
_DecodeResult changes

AnastasiyaTarasova committed:
bugfix

AnastasiyaTarasova committed:
Added ErrorCode(IntEnum)

AnastasiyaTarasova committed:
Added InvalidNonce decoding

AnastasiyaTarasova committed:
Revert "Added InvalidNonce decoding"

This reverts commit d04cb3134be75f20565470709f88dc97a5d8a128.

AnastasiyaTarasova committed:
Added dataclass_struct lib

AnastasiyaTarasova committed:
Added factory wip

AnastasiyaTarasova committed:
Added _get_evm_error_log_list

AnastasiyaTarasova committed:
Added Sequence

AnastasiyaTarasova committed:
Added error checking

AnastasiyaTarasova committed:
bugfix

AnastasiyaTarasova committed:
added logs

dependabot[bot] committed:
build(deps): bump aiohttp from 3.11.4 to 3.11.6

Bumps [aiohttp](https://github.com/aio-libs/aiohttp) from 3.11.4 to 3.11.6.
- [Release notes](https://github.com/aio-libs/aiohttp/releases)
- [Changelog](https://github.com/aio-libs/aiohttp/blob/master/CHANGES.rst)
- [Commits](https://github.com/aio-libs/aiohttp/compare/v3.11.4...v3.11.6)

---
updated-dependencies:
- dependency-name: aiohttp
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

AnastasiyaTarasova committed:
fix

afalaleev committed:
rpc: allow input in eth_estimateGas

AnastasiyaTarasova committed:
decode_old -> decode

neonlabstech commented:
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |  496304 |      0     |      0.412 |    0.00253766  |         0.001 |
| getSlot                           |  226923 |      0     |      0.421 |    0.0015884   |         0.001 |
| getMultipleAccounts               |  108141 |      0     |      0.397 |    0.00337419  |         0.001 |
| getTransaction                    |  104730 |      0     |      0.342 |    0.00985564  |         0.006 |
| getBlocks                         |   98954 |      0     |      0.204 |    0.00106735  |         0.001 |
| getLatestBlockhash                |   55863 |      0     |      0.472 |    0.00380132  |         0.001 |
| sendTransaction                   |   52606 |      0     |      0.468 |    0.0133487   |         0.008 |
| getBlockTime                      |   52399 |      0     |      0.17  |    0.00102815  |         0     |
| getBlock                          |   50836 |      0     |      0.082 |    0.00223643  |         0.002 |
| getBalance                        |   15217 |      0     |      0.148 |    0.00263929  |         0.001 |
| getSignatureStatuses              |    1437 |      0     |      0.194 |    0.00295825  |         0.001 |
| getVersion                        |     741 |      0     |      0.167 |    0.00284345  |         0.001 |
| isBlockhashValid                  |     700 |      0     |      0.025 |    0.000937143 |         0     |
| simulateTransaction               |     491 |      0     |      0.004 |    0.00125662  |         0.001 |
| getSignaturesForAddress           |      63 |      0     |      0.019 |    0.00338095  |         0.002 |
| requestAirdrop                    |      32 |      0     |      0.002 |    0.00096875  |         0.001 |
| getHealth                         |       3 |      0.001 |      0.026 |    0.011       |         0.006 |
| getTokenAccountBalance            |       3 |      0     |      0.001 |    0.000666667 |         0.001 |
| getFirstAvailableBlock            |       2 |      0     |      0     |    0           |         0     |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |         0.001 |
| getClusterNodes                   |       1 |      0.004 |      0.004 |    0.004       |         0.004 |

</details>

afalaleev committed:
core-api: generate unique ID for requests

afalaleev committed:
chore: polish field names

afalaleev committed:
chore: tx formating in proxy-cli

afalaleev committed:
rpc: fix return types

afalaleev committed:
tx-exec: fix stuck processing for another token space

AnastasiyaTarasova committed:
bugfix

vodolaz committed:
test infra instance size changed

vodolaz committed:
test infra instance size changed

AnastasiyaTarasova committed:
SOlTxListSender -> NeonSolTxListSender

AnastasiyaTarasova committed:
import fix

AnastasiyaTarasova committed:
decode fix

AnastasiyaTarasova committed:
NeonTxErrorParser fix

AnastasiyaTarasova committed:
amend

AnastasiyaTarasova committed:
resource_manager.py changes

AnastasiyaTarasova committed:
added _decode_tx_Status to tx_list_sender

AnastasiyaTarasova committed:
log fix

AnastasiyaTarasova committed:
SolTxIdx fix

AnastasiyaTarasova committed:
transaction_decoder from_raw fix

AnastasiyaTarasova committed:
transaction_list_sender fix

dependabot[bot] committed:
build(deps): bump aiohttp from 3.11.6 to 3.11.7

---
updated-dependencies:
- dependency-name: aiohttp
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump pydantic from 2.9.2 to 2.10.1

Bumps [pydantic](https://github.com/pydantic/pydantic) from 2.9.2 to 2.10.1.
- [Release notes](https://github.com/pydantic/pydantic/releases)
- [Changelog](https://github.com/pydantic/pydantic/blob/main/HISTORY.md)
- [Commits](https://github.com/pydantic/pydantic/compare/v2.9.2...v2.10.1)

---
updated-dependencies:
- dependency-name: pydantic
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

AnastasiyaTarasova committed:
resource_manager.py fix

AnastasiyaTarasova committed:
neon_tx_error_parser fix

AnastasiyaTarasova committed:
Revert "neon_tx_error_parser fix"

This reverts commit e8994bc023e9b88488201b74cf2674ac976aa141.

AnastasiyaTarasova committed:
Revert "resource_manager.py fix"

This reverts commit 24bee4140f3a1d3733008989b197f929c82209f4.

AnastasiyaTarasova committed:
evm_log_decoder refactoring

AnastasiyaTarasova committed:
neon_tx_error_parser refactoring

AnastasiyaTarasova committed:
strategy_base refactoring

AnastasiyaTarasova committed:
SolTxListSender -> NeonSolTxListSender

AnastasiyaTarasova committed:
transaction_list_sender refactoring

AnastasiyaTarasova committed:
transaction_decoder refactoring

afalaleev committed:
NDEV-3424 rpc: allow 'data' and 'input' properties in eth_estimateGas

AnastasiyaTarasova committed:
decode_old -> decode

AnastasiyaTarasova committed:
from_raw_old -> from_raw

AnastasiyaTarasova committed:
refactoring

AnastasiyaTarasova committed:
Revert "refactoring"

This reverts commit 1f8aec514cd58b70275c60172b2902de89836c88.

AnastasiyaTarasova committed:
Merge branch 'develop' into NDEV-3382-evm-should-return-error-codes

AnastasiyaTarasova committed:
refactoring

afalaleev commented on common/neon/evm_log_decoder.py:
```diff
@@ -173,6 +260,11 @@ def default(cls) -> Self:
+class SolTxIdx:
+    sol_tx_sig: SolTxSig
+    sol_ix_idx: int
+    sol_inner_ix_idx: int | None
```
This structure isn't used.
Remove it.

AnastasiyaTarasova replied:
removed https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/f8f4ab2a37ea671d5b96244f72dda5acbc0625df

afalaleev commented on common/neon/transaction_decoder.py:
```diff
@@ -108,6 +108,7 @@ class SolNeonTxIxMetaInfo:
 
     @classmethod
     def from_raw(cls, sol_tx: SolTxMetaInfo, sol_tx_ix: SolTxIxMetaInfo, sol_log: SolTxIxLogInfo) -> Self:
+
```
Remove this change

AnastasiyaTarasova replied:
done

afalaleev commented on common/neon/evm_log_decoder.py:
```diff
@@ -103,6 +103,92 @@ def log_bloom(self) -> int:
+
+    @classmethod
+    def from_raw(cls, code: int, data: bytearray, message: str):
+        return NeonTxErrorLogInfo(
```
```
return cls(
    ...
)
```

AnastasiyaTarasova replied:
done https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/f86851d1f6ccf155b35eb8e617a16e391aaa713a

afalaleev commented on common/neon/evm_log_decoder.py:
```diff
@@ -103,6 +103,92 @@ def log_bloom(self) -> int:
+            message = message,
+        )
+
+    def to_clean_copy(self, log: _NeonTxLogDraft) -> NeonTxErrorLogInfo:
```
The parameter `log` isn't used.
Remove it

AnastasiyaTarasova replied:
done

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+        error_log_list: list[NeonTxErrorLogInfo] = list()
+        for log_info in log_state.log_list:
+            if log_info.prog_id == NeonProg.ID:
+                log_list.extend(log_info.log_msg_list())
```
Just replace this copy-paste with the call `evm_log_list = self._get_evm_log_list()`

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+            sol_inner_ix_idx=None,
+            is_success=False,
+            _rpc_tx_ix=None,
+            _tx_acct_key_list=None)
```
Replace this with the `fake_tx_ix = SolTxIxMetaInfo.default()`
Like in this line:
https://github.com/neonlabsorg/neon-proxy.py/blob/7a63b175fc74c30c2ec80f168f5861f5cec47444/proxy/executor/strategy_base.py#L377-L382

AnastasiyaTarasova replied:
replaced https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/2378a85dce5da8d221ea792efb0bb790b4c36f6f

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+import re
+import logging
+from common.utils.cached import cached_method, cached_property
```
`cached_property` isn't used

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+import re
+import logging
+from common.utils.cached import cached_method, cached_property
+from common.solana.transaction_decoder import SolTxMetaInfo
```
`SolTxMetaInfo` isn't used.

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+    SolRpcNodeUnhealthyErrorInfo,
+    SolRpcTxReceiptInfo,
+    SolRpcInvalidParamErrorInfo,
+)
```
The following types aren't used:
```
  SolRpcTxIxErrorInfo,
    SolRpcTxIxFieldErrorCode,
    SolRpcTxErrorInfo,
    SolRpcTxFieldErrorCode,
    SolRpcNodeUnhealthyErrorInfo,
    SolRpcTxReceiptInfo,
    SolRpcInvalidParamErrorInfo,
```

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+    SolRpcInvalidParamErrorInfo,
+)
+from common.solana.log_tree_decoder import SolTxLogTreeDecoder
+from common.neon.evm_log_decoder import SolTxIdx, NeonTxErrorLogInfo
```
`SolTxIdx` isn't used.

AnastasiyaTarasova replied:
removed https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/f8f4ab2a37ea671d5b96244f72dda5acbc0625df

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+            if log_rec.code == NeonTxErrorLogInfo.ErrorCode.InvalidTransactionNonce:
+                address = log_rec.data[0:19]
+                state_tx_cnt = int.from_bytes(data[20:27])
+                tx_nonce = int.from_bytes(data[28:35])
```
what is the `data` here?

AnastasiyaTarasova replied:
fixed https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/51e7853fa1b576e11592898216eed5ad8c6df1be

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+        for log_rec in log_list:
+            if log_rec.code == NeonTxErrorLogInfo.ErrorCode.OutOfGas:
+                has_gas_limit = int.from_bytes(log_rec.data[0:31])
+                req_gas_limit = int.from_bytes(log_recdata[32:64])
```
what is the `log_recdata` here?

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+from common.solana_rpc.transaction_list_sender import SolTxListSigner
+from common.solana_rpc.transaction_list_sender import SolTxStatClient
+from common.solana_rpc.transaction_list_sender import SolWatchTxSession
+from common.solana_rpc.transaction_list_sender import SolTxSendState
```
Please replace 5 lines above with the:
``` from ..transaction_list_sender import (
     SolTxListSender,
     SolTxListSigner,
     SolTxStatClient,
     SolWatchTxSession,
     SolTxSendState,
)
```

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+from common.solana_rpc.transaction_list_sender import SolTxStatClient
+from common.solana_rpc.transaction_list_sender import SolWatchTxSession
+from common.solana_rpc.transaction_list_sender import SolTxSendState
+from ..solana.signature import SolTxSig
```
`SolTxSig` isn't used

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+from common.solana_rpc.transaction_list_sender import SolWatchTxSession
+from common.solana_rpc.transaction_list_sender import SolTxSendState
+from ..solana.signature import SolTxSig
+from ..solana.transaction import SolTx
```
`SolTx` isn't used

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+from common.solana_rpc.transaction_list_sender import SolTxSendState
+from ..solana.signature import SolTxSig
+from ..solana.transaction import SolTx
+from ..solana.transaction_meta import SolRpcTxSlotInfo, SolRpcTxReceiptInfo
```
`SolRpcTxSlotInfo` isn't used

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+from common.config.config import Config
+from common.solana_rpc.errors import (
+    SolUnknownReceiptError,
+    SolBlockhashNotFound,
```
`SolBlockhashNotFound` isn't used.

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+    SolUnknownReceiptError,
+    SolBlockhashNotFound,
+    SolNeonRequireResizeIterError,
+    SolCbExceededError,
```
`SolNoMoreRetriesError` isn't used.

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+            sol_session: SolWatchTxSession,
+            sol_tx_signer: SolTxListSigner,
+    ) -> None:
+        super().__init__(cfg, stat_client, sol_session, sol_tx_signer)
```
Surplus ctor, it doesn't do anything

AnastasiyaTarasova replied:
removed https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/e6a5ade4a18fc7e8c190ab7d13fcd802b0902a1e

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+        super().__init__(cfg, stat_client, sol_session, sol_tx_signer)
+
+    def clear(self) -> None:
+        super().clear()
```
Surplus method

AnastasiyaTarasova replied:
removed https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/e6a5ade4a18fc7e8c190ab7d13fcd802b0902a1e

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+    def clear(self) -> None:
+        super().clear()
+
+    def _add_tx_receipt(
```
This method - is just a copy-paste of the parent method

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+        self._tx_state_dict[tx_state.tx.sig] = tx_state
+        self._tx_state_list_dict.setdefault(tx_state.status, list()).append(tx_state)
+
+    def _decode_tx_status(self, tx: SolTx, now: int, tx_receipt: SolRpcTxReceiptInfo) -> _DecodeResult:
```
Not clear - why this method is copy-paste of the parent?

AnastasiyaTarasova replied:
separated the logic https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/2331d8425605e52dc036fd98edec75df66c4ad72

afalaleev commented on common/solana_rpc/transaction_error_parser.py:
```diff
@@ -164,28 +126,9 @@ def _get_tx_ix_error(self) -> SolRpcTxIxErrorInfo | None:
 
     @cached_method
-    def _get_log_list(self) -> Sequence[str]:
+    def _get_log_list(self) -> tuple[str, ...]:
```
For what?

afalaleev commented on common/solana_rpc/transaction_list_sender.py:
```diff
@@ -30,6 +30,8 @@
 from ..solana.transaction import SolTx
 from ..solana.transaction_meta import SolRpcTxSlotInfo, SolRpcTxReceiptInfo
 
+from common.neon.neon_tx_error_parser import NeonTxErrorParser
```
`from ..neon.neon_tx_error_parser`

AnastasiyaTarasova replied:
done https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/637e5fd8f343b9e37e853e19d5929b22ad39352c

afalaleev commented on common/solana_rpc/transaction_list_sender.py:
```diff
@@ -429,55 +429,55 @@ class _DecodeResult:
     def _decode_tx_status(self, tx: SolTx, now: int, tx_receipt: SolRpcTxReceiptInfo) -> _DecodeResult:
         status = SolTxSendState.Status
-        tx_error_parser = SolTxErrorParser(tx, tx_receipt)
+        neon_tx_error_parser = NeonTxErrorParser(tx, tx_receipt)
```
**Solana** Tx Sender shouldn't know about the `NeonTxErrorParser`

AnastasiyaTarasova replied:
fixed https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/f3e6944115db4f157649c2f51e11ffe93a5f93cd

afalaleev commented on common/solana_rpc/neon_sol_tx_list_sender.py:
```diff
@@ -0,0 +1,123 @@
+import dataclasses
```
`dataclasses` isn't used

afalaleev commented on common/neon/neon_tx_error_parser.py:
```diff
@@ -0,0 +1,116 @@
+import re
```
Looks bad that this file is in the `neon/`
Because there are cross includes between 2 directories `solana_rpc` and `neon`

AnastasiyaTarasova replied:
moved to solana_rpc https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/4b36c423c53496513e52e8e77875b5abf402a2a2

afalaleev commented on common/solana_rpc/transaction_error_parser.py:
```diff
@@ -25,19 +24,23 @@ class SolTxErrorParser:
+            if self._get_tx_ix_error() == SolRpcTxIxFieldErrorCode.ProgramFailedToComplete:
+                return True
+
+        log_list = self._get_evm_log_list()
```
This method is a Neon error, not Solana

AnastasiyaTarasova replied:
moved https://github.com/neonlabsorg/neon-proxy.py/pull/155/commits/a8fa9b90b14ee2f436c56b692b96d88a26aa4f0f

afalaleev commented on requirements.txt:
```diff
@@ -4,6 +4,7 @@ base58==2.1.1
 psycopg[binary,pool]==3.2.3
 hvac==2.3.0
 aioprometheus==23.12.0
+
```
For what?

afalaleev commented on tests/solana/test_cb_exceeded.py:
```diff
@@ -6,6 +6,7 @@
 from common.solana.transaction_meta import SolRpcTxSlotInfo
 from common.solana_rpc.transaction_error_parser import SolTxErrorParser
 from common.solana_rpc.transaction_list_sender import SolTxListSender
```
`SolTxListSender` isn't used

afalaleev commented on proxy/operator_resource/resource_manager.py:
```diff
@@ -24,6 +24,7 @@
 from common.solana.transaction import SolTx
 from common.solana.transaction_legacy import SolLegacyTx
 from common.solana_rpc.transaction_list_sender import SolTxListSender
```
`SolTxListSender` isn't used

AnastasiyaTarasova committed:
removed SolTxIdx

AnastasiyaTarasova committed:
neon_tx_error_parser refactoring dev

AnastasiyaTarasova committed:
syntax refactoring

AnastasiyaTarasova committed:
get_nonce_error fix

AnastasiyaTarasova committed:
syntax fix

AnastasiyaTarasova committed:
import fix

AnastasiyaTarasova committed:
syntax fix

AnastasiyaTarasova committed:
NeonTxErrorLogInfo to_clean_copy fix

AnastasiyaTarasova committed:
evm_log_decoder refactoring

AnastasiyaTarasova committed:
_get_evm_error_log_list refactoring

AnastasiyaTarasova committed:
moved check_if_require_resize_iter

AnastasiyaTarasova committed:
moved neon_sol_tx_list_sender.py to solana_rpc

AnastasiyaTarasova committed:
NeonSolTxListSender -> SolTxListSender in transacrion_list_sender.py

AnastasiyaTarasova committed:
import fix in neon_sol_tx_list_sender.py

AnastasiyaTarasova committed:
to_clean_copy fix (error otherwise)

AnastasiyaTarasova committed:
Dividing neon error handling and solana error handling

AnastasiyaTarasova committed:
refactoring neon_sol_tx_list_sender.py

AnastasiyaTarasova committed:
neon _decode_tx_status changes

afalaleev commented:
Close due to refactored PR https://github.com/neonlabsorg/neon-proxy.py/pull/229
